pubsub/internal/scheduler/publish_scheduler.go

```diff
@@ -106,20 +104,23 @@ func (s *PublishScheduler) Add(key string, item interface{}, size int) error {
 
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	b, ok := s.bundlers[key]
+	var b *bundler.Bundler
+	bInterface, ok := s.bundlers.Load(key)
+
 	if !ok {
-		s.outstanding[key] = 1
+		s.outstanding.Store(key, 1)
 		b = bundler.NewBundler(item, func(bundle interface{}) {
 			s.workers <- struct{}{}
 			s.handle(bundle)
 			<-s.workers
 
 			nlen := reflect.ValueOf(bundle).Len()
 			s.mu.Lock()
```

Comment:
So that's it! You left the lock between Add() and handle() so you don't load non-existing keys from the map inside the else clause. This gave me a hard time and I thought I was doing something wrong when I tried this.